tempesta_fw/str_simd.c

```diff
@@ -1042,6 +1066,89 @@ __tfw_strspn_avx2_128(const char *str, __m256i sm)
 
 	return r0 < 64 ? r0 : 64 + r1;
 }
+
+static inline size_t
+__tfw_match_ctext_vchar32(const char *str)
+{
+	unsigned int r;
+
+	__m256i v = _mm256_lddqu_si256((void *)str);
+
+	__m256i sub = _mm256_sub_epi8(v, __C.ZERO256);
+	__m256i x = _mm256_cmpeq_epi8(v, __C.DEL256);
+	x |= _mm256_cmpgt_epi8(__C.SP256, sub);
+	x ^= _mm256_cmpeq_epi8(v, __C.HTAB256);
```

Comment:
Also there is `_mm256_xor_si256()` function. Is it applicable here?

Reply:
Yes, this is exactly what the compiler does this for us.